### ISSUE TITLE
No different training time with more GPU

### ISSUE THREAD

**Issue by juunnn** [facebookresearch/wav2letter#476]
So I have 8 GTX 1080 GPUs , and trying to use them all for training ConvGLU. The first, I tring using only one GPU (using mpirun -n 1) the runtime for one epoch is  `runtime: 01:31:27`  but when I run with 6 GPUs runtime perepoch is about `runtime: 01:13:52` while the nvidia-smi command says I using 100% of 6 GPU Memories.
Am I doing it wrong? why there is no big difference from running with one GPU or more? Is there any better way to accelerate the training time?

**Comment by vineelpratap** [comment]
Hi, 
 Could you send a paste of the training logs.

**Comment by juunnn** [comment]
This is when I use 1 GPU, 
```
epoch:        1 | lr: 0.100000 | lrcriterion: 0.006000 | runtime: 01:27:21 | bch(ms): 902.01 | smp(ms): 0.38 | fwd(ms): 284.58 | crit-fwd(ms): 4.86 | bwd(ms): 576.69 | optim(ms): 26.93 | loss:   17.99178 | train-TER: 99.32 | train-WER: 100.07 | test.lst-loss:   17.27869 | test.lst-TER: 98.91 | test.lst-WER: 100.04 | avg-isz: 689 | avg-tsz: 101 | max-tsz: 955 | hrs:   44.54 | thrpt(sec/sec): 30.59
epoch:        2 | lr: 0.100000 | lrcriterion: 0.006000 | runtime: 01:27:56 | bch(ms): 907.97 | smp(ms): 0.35 | fwd(ms): 287.36 | crit-fwd(ms): 6.64 | bwd(ms): 580.47 | optim(ms): 26.50 | loss:   20.64979 | train-TER: 86.22 | train-WER: 99.94 | test.lst-loss:   16.11248 | test.lst-TER: 68.01 | test.lst-WER: 99.44 | avg-isz: 689 | avg-tsz: 101 | max-tsz: 955 | hrs:   44.54 | thrpt(sec/sec): 30.39
epoch:        3 | lr: 0.100000 | lrcriterion: 0.006000 | runtime: 01:27:55 | bch(ms): 907.92 | smp(ms): 0.35 | fwd(ms): 287.20 | crit-fwd(ms): 6.74 | bwd(ms): 580.30 | optim(ms): 26.48 | loss:   14.78624 | train-TER: 62.83 | train-WER: 98.35 | test.lst-loss:   12.41002 | test.lst-TER: 56.87 | test.lst-WER: 95.86 | avg-isz: 689 | avg-tsz: 101 | max-tsz: 955 | hrs:   44.54 | thrpt(sec/sec): 30.39
epoch:        4 | lr: 0.100000 | lrcriterion: 0.006000 | runtime: 01:28:04 | bch(ms): 909.44 | smp(ms): 0.35 | fwd(ms): 287.81 | crit-fwd(ms): 6.72 | bwd(ms): 581.19 | optim(ms): 26.48 | loss:   12.37665 | train-TER: 52.35 | train-WER: 94.88 | test.lst-loss:   10.78020 | test.lst-TER: 52.57 | test.lst-WER: 91.15 | avg-isz: 689 | avg-tsz: 101 | max-tsz: 955 | hrs:   44.54 | thrpt(sec/sec): 30.34
```
This is from using 6 GPUs

```
epoch:       92 | lr: 0.100000 | lrcriterion: 0.006000 | runtime: 01:36:39 | bch(ms): 1951.43 | smp(ms): 0.43 | fwd(ms): 371.01 | crit-fwd(ms): 8.41 | bwd(ms): 1539.82 | optim(ms): 28.64 | loss:   11.39610 | train-TER: 31.40 | train-WER: 70.97 | test.lst-loss:    4.90653 | test.lst-TER: 16.26 | test.lst-WER: 45.79 | avg-isz: 717 | avg-tsz: 087 | max-tsz: 134 | hrs:  142.20 | thrpt(sec/sec): 88.27
epoch:       93 | lr: 0.100000 | lrcriterion: 0.006000 | runtime: 01:40:45 | bch(ms): 2034.03 | smp(ms): 0.38 | fwd(ms): 378.20 | crit-fwd(ms): 8.59 | bwd(ms): 1616.13 | optim(ms): 28.30 | loss:    5.26870 | train-TER: 17.05 | train-WER: 48.09 | test.lst-loss:    3.67154 | test.lst-TER: 12.14 | test.lst-WER: 34.07 | avg-isz: 717 | avg-tsz: 087 | max-tsz: 134 | hrs:  142.20 | thrpt(sec/sec): 84.68
epoch:       94 | lr: 0.100000 | lrcriterion: 0.006000 | runtime: 01:28:02 | bch(ms): 1777.49 | smp(ms): 0.38 | fwd(ms): 362.07 | crit-fwd(ms): 8.25 | bwd(ms): 1376.88 | optim(ms): 27.49 | loss:    4.13270 | train-TER: 13.57 | train-WER: 38.87 | test.lst-loss:    3.03579 | test.lst-TER:  9.99 | test.lst-WER: 28.21 | avg-isz: 717 | avg-tsz: 087 | max-tsz: 134 | hrs:  142.20 | thrpt(sec/sec): 96.90
```
I use the same arch for reference, and half of the training data from when I use 1 GPU, Thank you.

**Comment by juunnn** [comment]
I just realize that the size of the training data is bigger when I use 6 GPUs, so it cannot be compared. I'm sorry for this mistake.

**Comment by joazoa** [comment]
The documentation states that wav2letter scales linearly (for big models) up to 64 gpus.
But as seen above:
1 gpu = ~30s/s
6 gpu = ~90s/s

Is the bottleneck in PCI-e ?

**Comment by vineelpratap** [comment]
Yes. You can also see that `bwd(ms)` is a very high compared to `fwd(ms)` which also indicates that bottleneck is when synchronizing gradients.
